### ISSUE TITLE
How to Link HTML and CSS Files Separately

### ISSUE THREAD

**Issue by Zh-hz2** [sparckles/Robyn#911]
I have a question about linking an external CSS file to my HTML. I need to separate my HTML and CSS files, and I'm unsure about the correct way to reference the CSS file in my HTML. I have not found any way to figure out yet.

Could you please guide me on how to use the <link> tag to connect my HTML file to the CSS file? What robyn function(s) can I use and how?

**Comment by sansyrox** [comment]
Hey @Zh-hz2 👋 

Do you want to serve a folder with HTML and CSS ?

**Comment by Zh-hz2** [comment]
Yes.

**Comment by Zh-hz2** [comment]
@sansyrox 
yes. actually I tried to put them in two folders but I failed, and tried to put them in one folder and also failed. I might need to properly call a function and variables that operates on paths. To call it correctly, I'm a bit confused about this. and I checked html tag <link>, and I cannot access to css file. I don't know if I lose something.  It might be better if you can offer a simple runnable example.

**Comment by SHI1992** [comment]
> 嘿 👋
> 
> 您想提供带有 HTML 和 CSS 的文件夹吗？

I believe it is necessary to implement a static file service, for example, in other web frameworks there is:


```
# Serve files in the static folder at the URL /static
app.static('/static', './static')
```

This allows you to directly reference various files in the ./static directory within HTML.
```
<link rel="stylesheet" href="../static/mycss.css"/>
<script type="text/javascript" src="../static/js/myjs.js"></script>
```
Regrettably, after going through the documentation, I haven't found how to implement a static file service.
I hope to be able to use it in the future.
Thank you.

**Comment by AswanthManoj** [comment]
> > 嘿 👋
> > 您想提供带有 HTML 和 CSS 的文件夹吗？
> 
> I believe it is necessary to implement a static file service, for example, in other web frameworks there is:
> 
> ```
> # Serve files in the static folder at the URL /static
> app.static('/static', './static')
> ```
> 
> This allows you to directly reference various files in the ./static directory within HTML.
> 
> ```
> <link rel="stylesheet" href="../static/mycss.css"/>
> <script type="text/javascript" src="../static/js/myjs.js"></script>
> ```
> 
> Regrettably, after going through the documentation, I haven't found how to implement a static file service. I hope to be able to use it in the future. Thank you.

Robyn does have an API for serving static files by the following method:

```python
from robyn import Robyn, serve_file

app = Robyn(__file__)

# You can serve a single file like
@app.get("/get_file")
async def serve(request):
    return serve_file(file_path="./static/image.png", file_name="image.png")


# Or for serving a directory containing static files
app.add_directory(route="/static", directory_path="./static")

app.start(port=8080)
```

**Comment by sansyrox** [comment]
Hey All 👋 

We just released v0.60.0 and renamed `add_directory` to `serve_directory` . https://robyn.tech/documentation/api_reference/file-uploads

Do have a look and let me know if you have any more questions.
